include/maliput_sparse/builder/builder.h

```diff
@@ -0,0 +1,387 @@
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+/**
+ * @file Builder API to construct a maliput::api::RoadGeometry.
+ *
+ * @details It allows simple construction of nested nodes in the maliput::api::RoadGeometry graph. The
+ * geometry details are expected to be computed / loaded outside this method to allow parallelization
+ * of that process. The graph construction is a synchronous operation though and requires
+ * that the geometry information is available when building maliput::api::Lanes.
+ *
+ * A super simple 2-lane dragway could be constructed as follows:
+ *
+ * @code{cpp}
+ * const maliput::math:Vector3 start_left_lane_1{0., 0., 0.};
+ * const maliput::math:Vector3 end_left_lane_1{100., 0., 0.};
+ * const maliput::math:Vector3 start_right_lane_1{0., 5., 0.};
+ * const maliput::math:Vector3 end_right_lane_1{100., 5., 0.};
+ * const maliput::math:Vector3 start_left_lane_2 = start_right_lane_1;
+ * const maliput::math:Vector3 end_left_lane_2 = end_right_lane_1;
+ * const maliput::math:Vector3 start_right_lane_2{0., 10., 0.};
+ * const maliput::math:Vector3 end_right_lane_2{100., 10., 0.};
+ *
+ * LineString3d left_line_string_1{start_left_lane_1, end_left_lane_1};
+ * LineString3d right_line_string_1{start_right_lane_1, end_right_lane_1};
+ * LineString3d left_line_string_2{start_left_lane_2, end_left_lane_2};
+ * LineString3d right_line_string_2{start_right_lane_2, end_right_lane_2};
+ *
+ * RoadGeometryBuilder rg_builder;
+ * std::unique_ptr<RoadGeometry> road_geometry = RoadGeometryBuilder()
+ *   .Id("two_lane_dragway")
+ *   .StartJunction()
+ *     .Id("j0")
+ *     .StartSegment()
+ *       .Id("j0_s0")
+ *       .StartLane()
```

Comment:
Wondering if we should provide a StartLeftLane and StartRightLane. In a way to suggest to the segment how the graph would be created independently of the geometry of the lanes.

Reply:
I don't think we should. Each new line will be indexed from 0 to +inf and following the maliput::api::Segment::index() docstring, that means the order should be set in increasing +r direction. I created extra details in the SegmentBuilder::StartLane() method to clarify this point.